drivers/overlay/joinleave.go

```diff
@@ -72,7 +87,7 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		}
 	}
 
-	d.peerDbAdd(nid, eid, ep.addr.IP, ep.mac,
+	d.peerDbAdd(nid, eid, ep.addr.IP, ep.addr.Mask, ep.mac,
```

Comment:
We can change the peer entry to carry `IPNet` instead of `IP` and `Mask` seperately so that we can always carry them together while sending through serf and while holding it in peerDb